core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -105,23 +105,15 @@ public String generateLoadBalancerEndpoint(Stack stack) {
     }
 
     public String getLoadBalancerUserFacingFQDN(Long stackId) {
-        Set<LoadBalancer> loadBalancers = loadBalancerPersistenceService.findByStackId(stackId);
-        if (!loadBalancers.isEmpty()) {
-            LoadBalancer preferredLB = loadBalancers.stream()
-                .filter(lb -> LoadBalancerType.PUBLIC.equals(lb.getType()))
-                .findAny()
-                .orElse(loadBalancers.iterator().next());
-            if (StringUtils.isNotEmpty(preferredLB.getFqdn())) {
-                return preferredLB.getFqdn();
-            } else {
-                return loadBalancers.stream()
-                    .map(LoadBalancer::getFqdn)
-                    .filter(StringUtils::isNotEmpty)
-                    .findFirst().orElse(null);
-            }
-        }
+        Set<LoadBalancer> loadBalancers = loadBalancerPersistenceService.findByStackId(stackId).stream()
+            .filter(lb -> StringUtils.isNotEmpty(lb.getFqdn()))
+            .collect(Collectors.toSet());
+        LoadBalancer loadBalancer = loadBalancers.stream()
+            .filter(lb -> LoadBalancerType.PUBLIC.equals(lb.getType()))
+            .findAny()
+            .orElse(loadBalancers.isEmpty() ? null : loadBalancers.iterator().next());
```

Comment:
you might use `orElseGet` in this case, like:
```
        LoadBalancer loadBalancer = loadBalancers.stream()
            .filter(lb -> LoadBalancerType.PUBLIC.equals(lb.getType()))
            .findAny()
            .orElseGet(() -> loadBalancers.stream().findAny().orElse(null));
```
it would prevent java from evaluating the code even when it's not necessary